sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClientIntegrationTest.java

```diff
@@ -128,17 +143,60 @@ void nonSessionMessageBatch(MessagingEntityType entityType) {
             .verifyComplete();
     }
 
+    /**
+     * Verifies that we can do following
+     * 1. create transaction
+     * 2. send message  with transactionContext
+     * 3. Rollback/commit this transaction.
+     */
+    @MethodSource("messagingEntityProviderWithTransaction")
+    @ParameterizedTest
+    void transactionMessageSendAndCompleteTransaction(MessagingEntityType entityType, boolean isCommit) {
+        // Arrange
+        setSenderAndReceiver(entityType, 0, false);
+        final boolean isSessionEnabled =  false;
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, false);
+
+        // Assert & Act
+        AtomicReference<ServiceBusTransactionContext> transaction = new AtomicReference<>();
+        StepVerifier.create(sender.createTransaction())
+            .assertNext(txn -> {
+                transaction.set(txn);
+                assertNotNull(transaction);
+            })
+            .verifyComplete();
+        assertNotNull(transaction.get());
+
+        // Assert & Act
+        StepVerifier.create(sender.send(message, transaction.get()))
+            .verifyComplete();
+        if (isCommit) {
+            StepVerifier.create(sender.commitTransaction(transaction.get()).delaySubscription(Duration.ofSeconds(1)))
+                .verifyComplete();
+            StepVerifier.create(receiveAndDeleteReceiver.receive().next())
+                .assertNext(receivedMessage -> {
+                    assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+                    messagesPending.decrementAndGet();
+                })
+                .verifyComplete();
+        } else {
+            StepVerifier.create(sender.rollbackTransaction(transaction.get()).delaySubscription(Duration.ofSeconds(1)))
```

Comment:
I'm not sure what you are testing here. There is only one send operation in this transaction. Transactions are multiple operations.

How did you verify the rollback worked?